### DISCUSSION TITLE
Anycubic Photon Mono X compatibility

### DISCUSSION THREAD

**Discussion by HelgeKeck** [sn4k3/UVtools#328] [General]
Hi there,

just want to mention that, even if not listed, the Photon Mono X is compatible with the per layer settings and advanced tools.

thanks for this nice piece of software

Helge Keck

**Comment by sn4k3** [comment]
I was told that Anycubic Monos are compatible, but bottom layers can't use overrides, is that the case? Can you test if any of the bottoms support different settings other than default?

**Reply by HelgeKeck** [reply]
well, it has the firmware transition layers, but anything other then that seems to work. i normaly set bottom layers to 0 and use for the bottom a layer range from 0-4 to set a longer exposure time, this eliminates the transition layers completely

**Reply by sn4k3** [reply]
How the transition layers work in that printer? What changes?

**Reply by HelgeKeck** [reply]
> How the transition layers work in that printer? What changes?

the printer overrides the for the first 10 normal layers the exposure time. if you set for example the first 4 bottom layers 20 seconds and the normal layers to 10 seconds, then the printer would use for frame #5 19 secs, #6 = 18, #7 = 17, ....

**Comment by HelgeKeck** [comment]
is there any special test you want me to run?

**Comment by Alde1980** [comment]
Hi I'm interested into use uv tools for my mono x prints and my intention is to follow [This guide from honza](https://blog.honzamrazek.cz/2022/02/a-step-by-step-guide-for-the-perfect-bed-adhesion-and-removing-elephant-foot-on-a-resin-3d-printer/?fbclid=IwAR3THMqAHsqZaQTQdy-unyEYw1THAw3LjRbEtQ1kwCwnYN-98LRiAMIbFpk) to get rid of elephant foot. 
My problem is I cannot get rid of firmware hardcoded first layers. I dont know how to set per layer custom exposure times and rest times too.

**Reply by sn4k3** [reply]
Anycubic machines are a pain...
But the enforced transition layers are only for exposure times right?
That mean if you run the script it will work, if the machine respect the wait/rest times set by the script

**Reply by HelgeKeck** [reply]
just dont use first layers at all, this way you can control everything without firmware BS

**Reply by sn4k3** [reply]
> just dont use first layers at all, this way you can control everything without firmware BS

You mean, set bottom layer count to 0?

**Reply by HelgeKeck** [reply]
yes

**Reply by sn4k3** [reply]
> yes

Ok, didn't know they allow 0 there! With that the transition layers are gone?

**Reply by Alde1980** [reply]
So as sn4k3 just set bottom layers to 0 or is a more elaborated process?

**Reply by HelgeKeck** [reply]
no, its the only way

**Reply by Alde1980** [reply]
Ok ty both will try ASAP

**Comment by gulbrillo** [comment]
I don't think this is true. While the Mono X can handle different exposures (after the bottom layers + 10 transition layers), it's unable to remain at one Z height, and it seems even unable to return to the same Z height - prints will fail immediately after the bottom layers. (maybe I'm doing something wrong?)

**Reply by sn4k3** [reply]
I don't have a machine to test, but i guess the problem is the relative position Z scheme they use.
If a relative position of 0mm is not supported because they cap that to a min/max and not allow 0 then it will never work. Chitu and others have a big trouble with zeroed values! They dont like it at all. On Creality (.cxdlp) a zero on wait time will crash the print, so the zero is really a bad guy for these guys

**Reply by HelgeKeck** [reply]
Im not sure what you exactly mean. Im doing this all the time, using zero bottom layers. if you do that you need of course take care of the exposure time of the first layers. But it defintely works

**Comment by printfoam** [comment]
@HelgeKeck -- curious if you could provide some guidance here...

Hi there -- two quick questions on over riding layer-by-layer settings.

If the bottom layers are set to zero:

(1) can the retract heights also be changed layer-by-layer? for example:

layer 1: retract height = 2mm,
layer 2: retract height = 10mm,
layer 3: retract height = 4 mm...
layer n: retract height = x mm
(2) is firmware version 0.2.3 compatible with this? or do I need to upgrade the firmware?

Thanks!
